### DISCUSSION TITLE
[V5] [Lucid] Select and filter relationship

### DISCUSSION THREAD

**Discussion by gideaoms** [adonisjs/core#2340] [Help] [answered]
Hi,

 I have a `table` called `products` that belongs to `stores` that belongs to a `users`. I want to get all products that belong to a specific user. I am doing it:
```ts
const products = await ProductModel.query()
  .preload('store', (query) => {
    query.preload('user', (query) => {
      query.where({ id: 'any id' })
    }
  }
```
The problem is that the `products` that do not belong to a specific user will be loaded anyway. My goal is to filter and get only the `products` that belongs to a specific user:
```ts
[
  {
    "id": "any product id",
    "description": "any product description",
    "store": {
      "id": "any store id",
      "name": "any store name"
    }
  }
]
```
Thanks.

**Comment by pk4all** [comment]
I am not sure but you can use like this,
```javascript
const products = await ProductModel.query()
                  .preload('store', (query) => {
                    query.preload('user', (query) => {
                      query.where({ id: 'any id' })
                    }).has('user','>',0)
                  })
```

**Reply by gideaoms** [reply]
Thank you but it did not work. All products that do not have stores are being returned.

**Reply by pk4all** [reply]
You can use same .has('store','>',0) in your query.

**Comment by RomainLanz** [comment, marked as answer]
Hey @gideaoms! 👋 

You have multiple solutions.

1. You can use `whereHas`.
 ➡️  https://preview.adonisjs.com/guides/model-relations/introduction#querying-relationship-existence

2. You can query through the `User` instead.

**Reply by popovicieduard** [reply]
Maybe using has many through ? https://preview.adonisjs.com/guides/model-relations/has-many-through

**Reply by gideaoms** [reply]
Thank you @RomainLanz 

I decided to use the first one.
